### PR TITLE
add a switch case for tidy code in main.cpp

### PR DESCRIPTION
**Pull request by 1ctinus**
I would recommend to check if the syntax is proper; but it looks fine to me.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -65,63 +65,64 @@ class SubchainModule : public EditProcessor {
 
 
 void addChainLink(EditProcessChain & procchain, const string & modulename, Setting & moduleconfig) {
-    if(modulename == "character_counts") {
+    switch (modulename) {
```
Comment (DamianZaremba) [inline src/main.cpp:68]: `modulename` is a `string` which cannot be switched.

Internally it is really a char array which doesn't simply match at compile time for building the branch tables.

File: src/main.cpp [hunk 2]
```diff
@@ -290,9 +291,9 @@ void addConfigChain(EditProcessChain & procchain, Setting & configchain, Setting
             string modulename = chainelname;
             if(linkcfg.exists("module")) modulename = (const char *)linkcfg["module"];
             addChainLink(procchain, modulename, linkcfg);
-        } else if(chaincfgs.exists(chainelname)) {    // Check if it's the name of another chain
+        break; if(chaincfgs.exists(chainelname)) {    // Check if it's the name of another chain
```
Comment (DamianZaremba) [inline src/main.cpp:294]: This logic is looping through each config section - breaking out after processing the first item will break the current logic.

`continue` might make sense here, however there are only 3 branches.

File: src/main.cpp [hunk 3]
```diff
@@ -388,7 +389,7 @@ int main(int argc, char **argv) {
                 }
 #ifdef SINGLETHREAD
                 chain.process(ed);
-#else
+#break;
```
Comment (DamianZaremba) [inline src/main.cpp:392]: This breaks the logic when compiled for multi threading since `tpool.submitEdit(ed);` will not be called.

**Review by DamianZaremba — CHANGES_REQUESTED** [review]
Code does not currently compile - you can use the `Dockerfile` to build the binary.

Left a number of inline comments.

**Comment by 1ctinus** [comment]
:/

**Pull request closed without merging**